spotbugs/src/test/java/edu/umd/cs/findbugs/detect/FindNonShortCircuitTest.java

```diff
@@ -0,0 +1,46 @@
+package edu.umd.cs.findbugs.detect;
+
+import edu.umd.cs.findbugs.BugInstance;
+import edu.umd.cs.findbugs.PrintingBugReporter;
+import edu.umd.cs.findbugs.Priorities;
+import edu.umd.cs.findbugs.Project;
+import edu.umd.cs.findbugs.ba.AnalysisContext;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class FindNonShortCircuitTest {
+
+    @Test
+    public void testBugTypeAndPriority() {

```

Comment:
Ideally this name should be more self-documenting, especially for such a long unit test; what behavior does it expect?

Reply:
Will  `testBugTypeAndPriorityDependOnSawBooleanFlags` name do?

I know this test is indecently long. 

But look at the terrible thing I had to do with static `AnalysisContext` just to run the test. So I decided to do it only once instead of four times. Having static state makes things non-testable, I think this is why I don't see any other non-trivial unit tests in `edu.umd.cs.findbugs.detect` package.

Reply:
Have you tried JUnit's `@Before` and `@After` annotations?

Reply:
@ThrawnCA Well you're right, I split the test.